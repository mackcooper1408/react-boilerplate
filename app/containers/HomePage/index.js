/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import LinkItem from 'components/LinkItem';
import H1 from 'components/H1';
import List from 'components/List';
import messages from './messages';

// items to be displayed | id used for unique key props
const items = [
  {
    id: '1',
    text: 'See the Strings',
    link: '/strings',
  },
  {
    id: '2',
    text: 'Add a String',
    link: '/add',
  },
];

export default function HomePage() {
  return (
    <div>
      <H1>
        <FormattedMessage {...messages.header} />
      </H1>
      <List items={items} component={LinkItem} />
    </div>
  );
}
