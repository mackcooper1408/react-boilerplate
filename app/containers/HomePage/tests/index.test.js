import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import HomePage from '../index';
import configureStore from '../../../configureStore';
import history from '../../../utils/history';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ConnectedRouter history={history}>
            <HomePage />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
