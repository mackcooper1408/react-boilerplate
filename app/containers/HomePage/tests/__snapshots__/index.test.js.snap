// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div>
  <h1
    class="H1-tncdg6-0 gNmgOQ"
  >
    <span>
      What Would You Like To Do?
    </span>
  </h1>
  <div
    class="Wrapper-sc-1umgotm-0 kPmOOE"
  >
    <ul
      class="Ul-mwnq6h-0 cbNGBL"
    >
      <li
        class="Wrapper-uulbip-0 cxfpqH"
      >
        <a
          class="Item-sc-50t3dd-0 gOsvcT"
          href="/strings"
        >
          See the Strings
        </a>
      </li>
      <li
        class="Wrapper-uulbip-0 cxfpqH"
      >
        <a
          class="Item-sc-50t3dd-0 gOsvcT"
          href="/add"
        >
          Add a String
        </a>
      </li>
    </ul>
  </div>
</div>
`;
